lib/vertica/query.rb

```diff
@@ -83,7 +83,8 @@ def handle_data_row(message)
   def handle_command_complete(message)
     if buffer_rows?
       @result = Vertica::Result.new(row_description: @row_description, rows: @buffer, tag: message.tag)
-      @row_description, @buffer = nil, nil
+      @row_description = nil
+      @buffer.clear if buffer_rows?
```

Comment:
`clear` was not good because the result `@buffer` is cleared here. We should set a new array as `@buffer = []`